File: src/site/wenlong/dimens/languages/LanguagesFactory.java
package site.wenlong.dimens.languages;

/**
 * 多语言工厂类
 *
 * @author : 郭文龙
 * @date : 2019-05-01  01:01
 */
public class LanguagesFactory {
    public static Text createText(int languagesEnum) {
        Text text = null;
        switch (languagesEnum) {
            case 1:
                text = Chinese.getInstance();
                break;
            case 0:
                text = English.getInstance();
                break;
            default:
                text = English.getInstance();
                break;
        }
        return text;
    }
}

File: src/main/kotlin/site/wenlong/dimens/languages/LanguagesFactory.kt
package site.wenlong.dimens.languages

/**
 * 语言工厂类
 *
 * @author : 郭文龙
 * @Email  : guowenlong20000@sina.com
 * @date   : 2023/5/7  0:55
 */
object LanguagesFactory {
    fun createText(languagesEnum: Int): Text {
        return when (languagesEnum) {
            1 -> Chinese
            0 -> English
            else -> English
        }
    }
}